typescript/sdk/src/token/app.ts

```diff
@@ -0,0 +1,35 @@
+import { TokenRouter } from '@hyperlane-xyz/core';
+import { objKeys } from '@hyperlane-xyz/utils';
+
+import { appFromAddressesMapHelper } from '../contracts/contracts';
+import { HyperlaneAddressesMap, HyperlaneContracts } from '../contracts/types';
+import { MultiProvider } from '../providers/MultiProvider';
+import { GasRouterApp } from '../router/RouterApps';
+
+import { HypERC20Factories, hypERC20factories } from './contracts';
+
+export class HypERC20App extends GasRouterApp<HypERC20Factories, TokenRouter> {
+  constructor(contractsMap: any, multiProvider: any) {
+    super(contractsMap, multiProvider);
+  }
+
+  router(contracts: HyperlaneContracts<HypERC20Factories>): any {
```

Comment:
I definitely sympathize that the types in these class hierarchies can be cumbersome but we should remain strict about the param and return types in function definitions. Forcing a cast inside the fn is less of an issue but strongly typed interfaces are worthwhile

Reply:
this "any" is hard where the the hypERC20Factories value types and the TokenRouter types don't line up.

Reply:
resolved